sdk/python/tests/nem/test_MessageEncoder.py

```diff
@@ -0,0 +1,81 @@
+import unittest
+
+from symbolchain.CryptoTypes import PrivateKey, PublicKey
+from symbolchain.nc import Message, MessageType
+from symbolchain.nem.KeyPair import KeyPair
+from symbolchain.nem.MessageEncoder import MessageEncoder
+
+from ..test.BasicMessageEncoderTest import BasicMessageEncoderTest
+
+
+class MessageEncoderTests(BasicMessageEncoderTest, unittest.TestCase):
+	KeyPair = KeyPair
+	MessageEncoder = MessageEncoder
+
+	def test_sender_can_decode_deprecated_encoded_message(self):
+		# Arrange:
+		key_pair = KeyPair(PrivateKey.random())
+		recipient_public_key = KeyPair(PrivateKey.random()).public_key
+		encoder = MessageEncoder(key_pair)
+		encoded = encoder.encode_deprecated(recipient_public_key, b'hello world')
+
+		# Act:
+		result, decoded = encoder.try_decode(recipient_public_key, encoded)
+
+		# Assert:
+		self.assertTrue(result)
+		self.assertEqual(decoded, b'hello world')
+
+	def test_recipient_can_decode_deprecated_encoded_message(self):
+		# Arrange:
+		key_pair = KeyPair(PrivateKey.random())
+		recipient = KeyPair(PrivateKey.random())
+		encoder = MessageEncoder(key_pair)
+		encoded = encoder.encode_deprecated(recipient.public_key, b'hello world')
+
+		# Act:
+		decoder = MessageEncoder(recipient)
+		result, decoded = decoder.try_decode(key_pair.public_key, encoded)
+
+		# Assert:
+		self.assertTrue(result)
+		self.assertEqual(decoded, b'hello world')
+
+	def test_decode_falls_back_to_input_cbc_fallback_has_wrong_padding(self):
+		# Arrange:
+		key_pair = KeyPair(PrivateKey.random())
+		recipient_public_key = KeyPair(PrivateKey.random()).public_key
+		encoder = MessageEncoder(key_pair)
+		encoded = encoder.encode_deprecated(recipient_public_key, b'hello world')
+
+		crafted_byte = encoded.message[-1] ^ 0xFF
+		encoded.message = encoded.message[:-1] + bytes([crafted_byte])
+
+		# Act:
+		result, decoded = encoder.try_decode(recipient_public_key, encoded)
+
+		# Assert:
+		self.assertFalse(result)
```

Comment:
shouldn't this test be True if payload is properly CBC encoded?

Reply:
this test has misleading name, see what next assert is checking
decode fails, cause padding is incorrect, so there's (False, original) returned

Reply:
i guess BasicMessageEncoderTest is handling GCM decode/encode?